src/deckgl/raster-layer/raster-layer-webgl2.fs.glsl

```diff
@@ -37,7 +37,7 @@ void main(void) {
 
   DECKGL_MUTATE_COLOR(image, vTexCoord);
 
-  color = apply_opacity(color_tint(color_desaturate(image.rgb)), opacity);
+  color = apply_opacity(color_tint(color_desaturate(image.rgb)), image.a * opacity);
```

Comment:
`image.a` won't always range from 0-1, will it?

Reply:
Even if input image.a is non-byte, there must be a module that transforms all non-byte values to bytes, right?

Reply:
I don't think so, because currently I just discard `image.a`. So for example, if you do NDVI and R and G are collapsed to -1 to 1 but A wouldn't be touched, and the values of A would be the same as the input values.

Reply:
A is overridden to 0 in NDVI: https://github.com/kylebarron/deck.gl-raster/blob/master/src/webgl/spectral-indices/normalized-difference.js#L8

Reply:
I still think that reverting to BitmapLayer's `image.a * opacity` after all modules are applied should be fine, because the last module in the chain is expected to convert input values to colors, so that it can be rendered to the user.

Otherwise, skipping a module to convert to colors would mean that the render target must be a float texture. Is there a usecase for it?

Reply:
> A is overridden to 0 in NDVI

Ah you're right. Though maybe in some cases it would be more general to do 

```
    image = vec4(normalized_difference_calc(image), image.g, image.b, image.a);
```

I don't know 🤷‍♂️